### ISSUE TITLE
Support out-of-bounds selection.

### ISSUE THREAD

**Issue by pthrasher** [atom/text-buffer#59]
In order for proper columnar selection to be implemented, you need to be able to highlight invalid regions. Right now all selection gets clipped to valid regions which makes implementing blockwise/columnar selection difficult.

Thoughts?

**Comment by maxbrunsfeld** [comment]
> In order for proper columnar selection to be implemented, you need to be able to highlight invalid regions. 

I don't think this feature is required for proper columnar selection. In vim's visual-block mode, there is no highlighting of invalid regions:

![screen shot 2015-05-26 at 5 39 12 pm](https://cloud.githubusercontent.com/assets/326587/7826392/2ff4ac3a-03ce-11e5-856c-ab5d778c3017.png)

I'm also not in favor of out-of-bounds selection in general. It's complex conceptually, as it means there are markers that don't correspond to any real region of text, and I've never seen it have any utility. Maybe could you describe some cases where it would be useful to you?

**Comment by pthrasher** [comment]
> I don't think this feature is required for proper columnar selection. In vim's visual-block mode, there is no highlighting of invalid regions:

Vim does support this if you use `set virtualedit+=block`

![2015-05-29 at 6 51 am](https://cloud.githubusercontent.com/assets/122682/7881314/b3fe5776-05cf-11e5-85df-2c2382222bbb.png)

> I've never seen it have any utility. Maybe could you describe some cases where it would be useful to you?

It's nice as a visual guide. Visual block mode is about drawing a block. With some blocks of text, it's not always simple (cognitively) to keep track of the goal column, and a visual indication make this less of a mental burden.

**Comment by pthrasher** [comment]
@maxbrunsfeld with the above comment of mine aside, even without out-of-bounds selection, we still need some way to save / restore goal column after recreating selections.

Any ideas on this? Should this be handled in text-buffer or atom-proper? If you could point me in the right direction, I'd be more than happy to take a stab at a PR.
